core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -912,22 +947,43 @@ private void updateSchema(final GraphConfig config) {
                         .build();
                 addSchema(newSchema);
             }
+
+            if (null != config.getLibrary() && config.getLibrary().exists(config.getGraphId())) {
+                // Set Schema if null.
+                final Pair<Schema, StoreProperties> pair = config.getLibrary().get(config.getGraphId());
+                schema = (null == schema) ? pair.getFirst() : schema;
+            }
         }
 
-        private void updateStoreProperties(final GraphConfig config) {
-            StoreProperties mergedStoreProperties = null;
+        private void updateStoreProperties(final GraphConfig config, final Store store) {
+            final StoreProperties mergedStoreProperties = new StoreProperties();
+
+            //Add Properties from ParentProperties
             if (null != parentStorePropertiesId) {
-                mergedStoreProperties = config.getLibrary().getProperties(parentStorePropertiesId);
+                mergedStoreProperties.merge(config.getLibrary().getProperties(parentStorePropertiesId));
             }
 
+            //Add Properties from Builder
             if (null != properties) {
-                if (null == mergedStoreProperties) {
-                    mergedStoreProperties = properties;
-                } else {
-                    mergedStoreProperties.merge(properties);
+                mergedStoreProperties.merge(properties);
+            }
+
+            //Add Properties from Store
+            if (null != store) {
+                final StoreProperties storeProperties = store.getProperties();
+                if (null != storeProperties && mergedStoreProperties.getProperties().isEmpty()) {
```

Comment:
What is the logic behind only adding the store properties if the `mergedStoreProperties` is empty? Does that mean you only want to add the store props if there are no parent or build props? Same goes for the Library bit on line 980.

Reply:
This has now been removed because it caused complexity with Kafka tests later down the line. these changes was reverted. but should be examined in the future.